### ISSUE TITLE
Postgres: consider excluding queries related to the pg adapter's schema and type caches

### ISSUE THREAD

**Issue by fallwith** [newrelic/newrelic-ruby-agent#1014]
The Postgres adapter seems to fire off 3 SELECT statements to ready its internal schema and type caches. These are currently being tallied by the agent as SELECT statements performed by the monitored application during normal operation. Seeing as they are SQL statements being readied and executed from the Ruby code base against the Postgres DB, perhaps it is desired that these queries are tallied. But given that they are expected for all apps, it may be appropriate to exclude them. We should discuss this and look at what others do (does Rails ignore these queries when logging, for example).

Here is what a quick `caller` based hack would look like to exclude such queries from being counted.

https://github.com/newrelic/newrelic-ruby-agent/pull/1004/commits/84cb15c2f08a54574a113f2738ca29ee674245fb

<img width="921" alt="Screen Shot 2022-03-07 at 11 58 59 AM" src="https://user-images.githubusercontent.com/121392/157108608-4732d1fe-6929-4dbc-80b0-132c726bf2b3.png">

**Comment by fallwith** [comment]
If we want to consider these queries as coming from the Ruby application, then perhaps we at least make the counting of these queries a configurable setting.

**Comment by fallwith** [comment]
Closing this one out as "won't fix" / research complete.

- ActiveRecord notifications and other APM based instrumentations do not appear to differentiate between query cache `SELECT` statements and other `SELECT` statements.
- As originally mentioned, these queries do come from the Ruby application being observed, so it's completely reasonable to accept them as application level query data that ought to be reported.
- Our own thoughts raised here stemmed from CI based observations. With our CI we are constantly freshly re-prepping ActiveRecord based connections. Long running customer web applications will only initialize query cache once at startup. A re-prep from scaling or deployment event would be something I as a customer would see value in having recorded.
- Waiting for a signal before starting instrumentation and getting into the business of interpreting `SEQUEL` statement intentions are both things that are potentially fraught with danger.
- We do not have any related feature requests that this work would address.
